core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java

```diff
@@ -498,7 +509,10 @@ protected void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) thro
                 // ignore
                 return;
             }
-            if (!NetworkExceptionHelper.isCloseConnectionException(e.getCause())) {
+            if (e.getCause() instanceof CircuitBreakingException) {
```

Comment:
If we think that circuit breaking should have a specific rest status, then we should do it in the CircuitBreakerException not here. To me saying the request too big (for any circuitbreaking, but that can be solved) is misleading. I'd prefer server unavailable.